test/studies/mstrout/wordCount.chpl

```diff
@@ -0,0 +1,178 @@
+/* wordCount.chpl
+   
+  Purpose:
+    For all of the files in a given directory, does a wordcount per file.
+    An example of using Chapel for processing files in parallel.  This
+    same code can do multi-core and/or distributed memory + multicore parallelism.
+
+  Usage:
+    Clone the main branch of the chapel compiler.
+      > git clone https://github.com/chapel-lang/chapel.git
+
+    Set up the environment to build the compiler and run Chapel programs
+      > cd chapel
+      > source util/setchplenv.bash
+
+    Build the Chapel compiler, this will take awhile...
+      > make
+
+    Compile this example
+      > cd test/studies/mstrout/
+      > chpl wordCount.chpl
+
+    Run this example
+      > ./wordCount
+
+    Doing wordCount on a different directory of files
+      > ./wordCount --InputDir="path-to-diff-directory"
+
+    Doing wordCount using a different number of threads per node
+      > CHPL_RT_NUM_THREADS_PER_LOCALE=1 ./wordCount
+      > CHPL_RT_NUM_THREADS_PER_LOCALE=2 ./wordCount
+      > CHPL_RT_NUM_THREADS_PER_LOCALE=4 ./wordCount
+
+    Running on multiple locales:
+      See the notes at the end of this file.
+
+  Credits:
+    Mostly based on https://github.hpe.com/michelle-strout/CodeExamples/tree/main/Chapel/DataAnalysisExample/final-word-count.chpl,
+    which was heavily edited by Ben McD.
+*/
+use IO;
+use List;
+use Map;
+use FileSystem;
+use Time;
+use BlockDist;
+
+config const inputDir = "InputFiles";
+config const debug = true;
+config const minCount = 2;      // minimum count for a word to show up in output
+
+//-----------------------------------------------------------------------
+// Read in the input csv files and at the same time do a wordcount per csv file.
+
+var t: Timer;
+t.start();
+
+var filenamesList: list(string);
+
+// findfiles() returns an iterator so putting all file names in a list
+// to enable creating an array of file names.
+for f in findfiles(inputDir) {
+    filenamesList.append(f);
+}
+// Create an array blocked into pieces per locale.
+var filenames = newBlockArr(0..#filenamesList.size, string);
+filenames = filenamesList;
+
+// Execute using distributed parallelism across locales if executing
+// using multilocales and within each locale in parallel.
+forall (i, file) in zip(filenames.domain, filenames) {
+  // Open up the input file.
+  var f = open(file, iomode.r);
+  if debug then writeln("\n\n====== filename = ", file);
+
+  // Create a reader of the input csv file
+  var reader = f.reader(locking=false);
+
+  // Read the first line to get the column names.
+  var line : string;
+  if (!reader.readline(line)) then
+    writeln("ERROR: ", file, " appears to be empty");
+
+  // declare a Map to maintain the count
+  var wordCount = new map(string,int);
+
+  // Read in all of the lines of the file.
+  while reader.readline(line) {
+    // count each time a word appears in the abstract column of this line
+    for word in line.split(" ") {
+      wordCount[word] += 1;
```

Comment:
This is including the newlines as part of the "word", which is why the output appears to have line breaks after "some", "count.", "we", and "text.".  (The newlines are emitted as part of the word, before the count of that "word".)

Something's counting an empty string, as evidenced by the start of the output line

```
wordCount = {: 12,
```

Maybe that's from having two spaces between sentences in the input?

Reply:
Another nice catch.  Should be fixed in the version I just pushed.